askar-storage/src/backend/db_utils.rs

```diff
@@ -453,6 +453,17 @@ pub trait QueryPrepare {
         }
         query
     }
+
+    fn order_by_query<'q>(mut query: String, order_by: Option<String>, descending: bool) -> String {
+        if let Some(order_by) = order_by {
+            query.push_str(" ORDER BY ");
+            query.push_str(&order_by);
```

Comment:
This order_by parameter needs to be sanitized, we don't want the caller to be able to inject raw SQL. Perhaps there should be an enum of allowed values? And do we want a default order (the ID I suppose)? In general the parameter might be a bit misleading as you can't sort by encrypted categories or values.

Reply:
Great points.

> This order_by parameter needs to be sanitized, we don't want the caller to be able to inject raw SQL. Perhaps there should be an enum of allowed values?

I think an enum would be appropriate if the set of allowed values is known and the same across all different record types. I don't know yet what that possible set of values would be, but I can review the DB.

> And do we want a default order (the ID I suppose)? 

Default ordering by ID is probably a good idea

> In general the parameter might be a bit misleading as you can't sort by encrypted categories or values.

This is correct -- it is a bit misleading if one can't sort by encrypted record values. Part of the motivation for this implementation was to at least allow users to query by "last updated" or "created" timestamps. I hope those will be possible. But if they're encrypted then naturally it'll just have to be explained as a note in documentation, why it's not possible to order by other fields.

I'll take some time to think about best steps forward.

Reply:
@andrewwhitehead I've implemented an Enum for OrderBy, and I noticed that it really only seems to make sense to ever order by ID. Please let me know if there are other fields that are generally safe to order by, which I can include in the enum. But for the time being, I think just ordering by ID is sufficient, to ensure consistency across multiple calls.

I've opted to preserved "backward compatibility" for now: if no OrderBy option is provided, there is no ordering, and the `descending` flag will also be ignored. I figured this is safer since I'm not sure if ID is a global field, in absolutely all records, so didn't want to force it to be always be default. If it is global, perhaps it's wise to make it default, even when no order by option is provided.

Reply:
I also ensured that limit, offset, order_by, and descending, will only ever be added to SELECT queries. So that the `extend_query` changes don't impact DELETE or INSERT query types, where pagination doesn't make sense.